### PR TITLE
Use completion API

### PR DESCRIPTION
**Pull request by DustinCampbell**
This change updates the existing AutoComplete end point to use the Roslyn Completion API, which allows the old keyword recommender code to be deleted. In order to properly support the Roslyn Completion API and all of its features, we'll need a new endpoint. I'll be working on that next.

cc @david-driscoll, @troydai

Note: The last commit is huge and is mostly around deleting code. If you want to review the meatier changes, take a look at the earlier commits.

### CODE REVIEW

**Comment by DustinCampbell** [comment]
Looks like dotnet restore is failing on CI builds. Does anyone know anything about that? @troydai?

**Comment by troydai** [comment]
Looking at it

**Comment by troydai** [comment]
NuGet references are out-of-date. I'm going to update them,.

**Comment by DustinCampbell** [comment]
Yeah, I just cleared my NuGet caches and can reproduce locally.

**Comment by troydai** [comment]
Sending out PR to address this: https://github.com/OmniSharp/omnisharp-roslyn/pull/584

You can rebase your changes onto my fix.

**Comment by DustinCampbell** [comment]
Yay! The CI builds have passed. Just need some code reviews.

File: src/OmniSharp.Abstractions/Configuration.cs [hunk 1]
```diff
@@ -3,7 +3,13 @@ namespace OmniSharp
...
         public static bool ZeroBasedIndices = false;
-        public const string CodeAnalysisVersion = "1.3.0.0";
-        public const string CodeAnalysisPublicKeyToken = "31bf3856ad364e35";
+
+        public const string RoslynVersion = "1.3.0.0";
+        public const string RoslynPublicKeyToken = "31bf3856ad364e35";
+
+        public static string GetRoslynAssemblyFullName(string name)
+        {
+            return $"{name}, Version={RoslynVersion}, Culture=neutral, PublicKeyToken={RoslynPublicKeyToken}";
```
Comment (troydai) [inline src/OmniSharp.Abstractions/Configuration.cs]: Issue filed: https://github.com/OmniSharp/omnisharp-roslyn/issues/585

**Comment by troydai** [comment]
LGTM

**Comment by DustinCampbell** [comment]
Awesome! @david-driscoll, when you get a chance, could you take a look? It's a pretty big change, so I want to be sure you're OK with it.

**Comment by david-driscoll** [comment]
:shipit:

**Comment by DustinCampbell** [comment]
Thanks!

**Pull request merged**